src/blocked.c

```diff
@@ -577,8 +577,12 @@ void handleClientsBlockedOnKeys(void) {
                     serveClientsBlockedOnStreamKey(o,rl);
                 /* We want to serve clients blocked on module keys
                  * regardless of the object type: we don't know what the
-                 * module is trying to accomplish right now. */
-                serveClientsBlockedOnKeyByModule(rl);
+                 * module is trying to accomplish right now.
+                 *
+                 * Optimization: If no clients are in type BLOCKED_MODULE,
+                 * we can skip this loop. */
+                if (server.blocked_clients_by_type[BLOCKED_MODULE])
+                    serveClientsBlockedOnKeyByModule(rl);
```

Comment:
Do we have any idea if it's worth doing this check for the other serveClientBlockedOn* functions? I'm guessing not, but it might be a minor optimization in some cases.

Also a minor nitpick: I would move the check into serverClientBlockedOnKeyByModule(). If anyone else calls it, it makes more sense there.

Reply:
Regarding `serveClientBlockedOn*` -- none of those showed up in our workload as additional overhead. I believe because we're switching on the type of the key, it will always pick the right one anyway.

Regarding the placement of the `if` -- the main reason I kept it on the outside was to save the function call. But chances are, it's negligible. Let me run another benchmark to see if it even shows up.

Reply:
@madolson I've verified that the function call overhead is negligible, and moved the check down into `serverClientBlockedOnKeyByModule`. 👍

Reply:
Thanks, it's a small thing but we've missed a couple places where we only optimize one of multiple call sites. Make it work, make it right, make it fast :)

I think serveClientBlockedOn* might be interesting to try to optimize, but you're right it's not a big time sync today.